bin/create-release

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env ruby
+require_relative '../deployment.rb'
+require 'cdo/only_one'
+require 'cdo/cdo_cli'
+include CdoCli
+
+require 'octokit'
+require 'open3'
+require 'tempfile'
+
+REPO = 'code-dot-org/code-dot-org'
+
+def git_generate_changelog
+  changes, _err, status = Open3.capture3("git log --merges --oneline production^1..production^2 --format=\"%s %b\"")
+
+  # command will error out unless the latest commit is a merge
+  return nil if status.exitstatus != 0
+
+  re = /Merge pull request #(?<pr>\d*) from code-dot-org\/(?<branch>\S*) (?<description>.*)/
+
+  matches = changes.split("\n").map do |change|
+    # first, extract the relevant data from the commit message
+    re.match(change)
+  end
+
+  relevant_matches = matches.select do |match|
+    # ignore merges from staging; those represent DTTs
+    match['branch'] != 'staging'
+  end
+
+  items = relevant_matches.map do |match|
+    # finally, create a changelog entry linking the description to the PR
+    "- [#{match['description']}](https://github.com/#{REPO}/pull/#{match['pr']})"
+  end
+
+  items.join("\n")
+end
+
+def get_next_release_version
+  # version format is "v{year}-{month}-{day}.{version number}"
+  today = Time.now.strftime("v%Y-%m-%d")
+  version_number = 0
+
+  # if we've already released a .0 or more today, increment the version
+  # number appropriately
+  last_version = Octokit.latest_release(REPO).tag_name
+  if last_version.start_with?(today)
+    matched_version = /v\d{4}-\d{2}-\d{2}\.(\d*)#/.match(last_version)
+
+    if matched_version.nil?
+      # The last release didn't have a valid version number; it could
+      # have been edited or created though some other route. In this
+      # rare case, we resort to guess-and-check
+
+      begin
+        # try no more than five versions before accepting that something
+        # else is horribly wrong
+        until version_number == 5
+          Octokit.release_for_tag(REPO, "#{today}.#{version_number}")
+          version_number += 1
+        end
+      rescue Octokit::NotFound
+        # intentional noop
+      else
+        puts "Tried and failed to find a valid version number for this date; please check that the latest release has an integer version number"
+        exit(-1)
+      end
+    else
+      version_number = matched_version[0].to_i + 1
+    end
+
+  end
```

Comment:
You've still got a bad case here:

Suppose a release history that looks like this (reverse chronological) and today is january 25.

```
someManuallyCreatedRelease
v2017-01-25.1
v2017-01-25.0
```

Seems like the correct next version here would either be `v2017-01-25.2` or `v2017-01-25.3`, because `last_version` doesn't start with today's date at all this helper returns `v2017-01-25.0` which results in a conflict.